pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -1322,13 +1322,16 @@ func (lbaas *LbaasV2) EnsureLoadBalancerDeleted(clusterName string, service *v1.
 		}
 	}
 
-	// delete all monitors
-	for _, monitorID := range monitorIDs {
-		err := v2monitors.Delete(lbaas.lb, monitorID).ExtractErr()
-		if err != nil && !isNotFound(err) {
-			return err
+	if lbaas.opts.CreateMonitor {
+
+		// delete all monitors
+		for _, monitorID := range monitorIDs {
+			err := v2monitors.Delete(lbaas.lb, monitorID).ExtractErr()
+			if err != nil && !isNotFound(err) {
```

Comment:
In https://github.com/kubernetes/kubernetes/issues/56313, you mention a log is returned `error log: resource not found`. Would you happen to have that log? This was checking `IsNotFound` before so it shouldn't have been returning an error here if it was that.

Reply:
error log:
==========
E1127 09:41:36.612951    1500 servicecontroller.go:753] Failed to process service. Retrying in 5s: Resource not found
E1127 09:41:42.783137    1500 servicecontroller.go:753] Failed to process service. Retrying in 10s: Resource not found
E1127 09:41:53.724673    1500 servicecontroller.go:753] Failed to process service. Retrying in 20s: Resource not found
E1127 09:42:14.857164    1500 servicecontroller.go:753] Failed to process service. Retrying in 40s: Resource not found
E1127 09:42:55.325303    1500 servicecontroller.go:753] Failed to process service. Retrying in 1m20s: Resource not found
=========================
Using LBaaS extension v2,  so I add this code to check monitor

Reply:
Emm... Have you read the codes from L1307, if the monitor is not found, it also not found in monitorIDs. So I can't understand why we need to add these codes. OTH, It may be another reason.

Reply:
I use v1.7.9 
-----
	for _, listener := range listenerList {
		pool, err := getPoolByListenerID(lbaas.network, loadbalancer.ID, listener.ID)
		if err != nil {
			return fmt.Errorf("Error getting pool for listener %s: %v", listener.ID, err)
		}
		poolIDs = append(poolIDs, pool.ID)
		monitorIDs = append(monitorIDs, pool.MonitorID)
	}
------
here not check pool.MonitorID is "".  I will close this. I'm sorry about this.